src/core/Akka/Actor/ReceiveActor.cs

```
@@ -20,10 +19,51 @@ namespace Akka.Actor
     public abstract class ReceiveActor : UntypedActor, IInitializableActor
     {
         private bool _shouldUnhandle = true;
-        private readonly Stack<MatchBuilder> _matchHandlerBuilders = new();
-        private PartialAction<object> _partialReceive = _ => false;
+        private readonly Stack<HandlerSet> _handlersStack = new();
+        private HandlerSet _currentHandlers = null;
         private bool _hasBeenInitialized;
 
+
+        // Internal classes to do the handling. If this approach is used then these can be
+        // Moved out and better tested
+        class HandlerSet
+        {
+            public HandlerSet()
```

The idea here is having something that handles the calls for each of the states of the actor.
1. TypedHandlers - Handlers when we have the type information
2. HandleObject - Handles calls when the type information supplied is object
3. Handle Any - The catch all handler 

Now that I am reviewing this I feel like the naming should be  a bit more consistent